Chore/162727542/convert database to postgres

habibaudu opened the pull request:
#### What does this PR do?
Convert mongo database to postgres and sequelize

#### Description of Task to be completed?
- create migrations folder and its migration file
- create controllers to register a user, log in using a passport, get User and update user
- change index.js to app.js
- create user model
- create config.json to configure Postgres database 
#### How should this be manually tested?
- fire up postman
   url : POST http://localhost:3000/api/v1/users
        email, username, password
   url : get http://localhost:3000/api/v1/user/:userId
        
   
#### Any background context you want to provide?
   None
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162727542
### Screenshots (if appropriate)
None
#### Questions:None

habibaudu committed:
chore(postgres):create controllers
create models
create config file
create migration files
update routes folder
[#162727542]

habibaudu committed:
chore(convert-to-postgres):fixing some violations
[#162727542]

habibaudu committed:
chore(convert-to-postgres):fix some violation
[#162727542]

habibaudu committed:
chore(convert-to-postgres):fix bug on update user controllerfix some violation
[#162727542]

oaboluwarin commented on app.js:
```diff
@@ -0,0 +1,47 @@
+import express from 'express';
+import bodyParser from 'body-parser';
```
Uninstall the `bodyParser` module.

Express now incorporates the `bodyParser` functionality itself. As such, the following will become:

`app.use(express.urlencoded({ extended: false }));`
`app.use(express.json());`

habibaudu replied:
Okay. will do that now

habibaudu replied:
I have uninstall the body-parser

habibaudu committed:
chore(convert-to-postgres):convert index.js to es6 to fix violations
[#162727542]

habibaudu committed:
chore(convert-to-postgres):uninstall body-parser
[#162727542]

oreoluwade commented:
Create a file `server.js` in the root directory. It is in this file that you should start the server. In essence, the `app` should be imported in there and these lines should be there:


`const server = app.listen(process.env.PORT || 3000, () => {
  console.log('Listening on port ' + server.address().port);
});`

habibaudu committed:
refactor routes
[#162727542]

habibaudu committed:
[#162727542] add eslint rules, update gitignore

habibaudu committed:
[#162727542] update config file, update controllers import

habibaudu committed:
[#162727542] fix eslint violations

habibaudu committed:
[#162727542] deprecate controller directory

habibaudu committed:
[#162727542] fix linting problems

habibaudu committed:
[#162727542] fix user model eslint problems

habibaudu committed:
[#162727542] cleanup
- convert config to js file
- refactor models index file
- add env to gitignore
